edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/stream/LogsStreamRequestHandler.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.IoTHub.Stream
+{
+    using System;
+    using System.Net.WebSockets;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.Logs;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class LogsStreamRequestHandler : IStreamRequestHandler
+    {
+        readonly ILogsProvider logsProvider;
+
+        public LogsStreamRequestHandler(ILogsProvider logsProvider)
+        {
+            this.logsProvider = Preconditions.CheckNotNull(logsProvider, nameof(logsProvider));
+        }
+
+        public async Task Handle(IClientWebSocket clientWebSocket, CancellationToken cancellationToken)
+        {
+            try
+            {
+                LogsStreamRequest streamRequest = await this.ReadLogsStreamingRequest(clientWebSocket, cancellationToken);
+                Events.RequestData(streamRequest);
+
+                var logOptions = new ModuleLogOptions(streamRequest.Id, LogsContentEncoding.None, LogsContentType.Text);
+                using (var socketCancellationTokenSource = new CancellationTokenSource())
+                {
+                    Task ProcessLogsFrame(ArraySegment<byte> bytes)
+                    {
+                        if (clientWebSocket.State != WebSocketState.Open)
+                        {
+                            Events.WebSocketNotOpen(streamRequest.Id, clientWebSocket.State);
+                            socketCancellationTokenSource.Cancel();
+                            return Task.CompletedTask;
+                        }
+                        else
+                        {
+                            return clientWebSocket.SendAsync(bytes, WebSocketMessageType.Binary, true, cancellationToken);
+                        }
+                    }
+
+                    using (var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, socketCancellationTokenSource.Token))
+                    {
+                        await this.logsProvider.GetLogsStream(logOptions, ProcessLogsFrame, linkedCts.Token);
+                    }
+                }
+
+                Events.StreamingCompleted(streamRequest.Id);
+            }
+            catch (Exception e)
+            {
+                Events.ErrorHandlingRequest(e);
+            }
+        }
+
+        async Task<LogsStreamRequest> ReadLogsStreamingRequest(IClientWebSocket clientWebSocket, CancellationToken cancellationToken)
+        {
+            var buf = new byte[1024];
```

Comment:
This is ensuring that each websocket frame is at most 1kb?

Reply:
This is simply the input payload that includes the details of the log request. I wanted to keep it below a fixed size. 1k seemed enough.

Reply:
This is to parse the first request on the socket?

Reply:
We can do this outside of this code review, but we need to go over the request format and get it documented. This is addition to the public API.

Reply:
Yes, this is indeed to parse the first request. 
The payload format - yes we need to document it. I have added it to the same doc on Edge Logging. I will get it reviewed with all.